src/freenet/client/filter/WAVFilter.java

```diff
@@ -0,0 +1,124 @@
+		public int wBitsPerSample = 0;
+		public int format = 0;
+	}
+
+	@Override
+	protected Object createContext() {
+		return new WAVFilterContext();
+	}
+
+	@Override
+	protected void readFilterChunk(byte[] ID, int size, Object context, DataInputStream input, DataOutputStream output,
+			String charset, Map<String, String> otherParams, String schemeHostAndPort, FilterCallback cb)
+			throws DataFilterException, IOException {
+		WAVFilterContext ctx = (WAVFilterContext)context;
+		if(ID[0] == 'f' && ID[1] == 'm' && ID[2] == 't' && ID[3] == ' ') {
+			if(ctx.hasfmt) {
+				throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "Unexpected fmt chunk was encountered");
+			}
+			if(size != 16 && size != 18 && size != 40) {
+				throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "fmt chunk size is invalid");
+			}
+			ctx.format = Short.reverseBytes(input.readShort());
+			if(ctx.format != WAVE_FORMAT_PCM && ctx.format != WAVE_FORMAT_IEEE_FLOAT && ctx.format != WAVE_FORMAT_ALAW && ctx.format != WAVE_FORMAT_MULAW) {
+				throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "WAV file uses a not yet supported format");
+			}
+			ctx.nChannels = Short.reverseBytes(input.readShort());
+			output.write(ID);
+			writeLittleEndianInt(output, size);
+			output.writeInt((Short.reverseBytes((short) ctx.format) << 16) | Short.reverseBytes((short) ctx.nChannels));
+			ctx.nSamplesPerSec = readLittleEndianInt(input);
+			writeLittleEndianInt(output, ctx.nSamplesPerSec);
+			int nAvgBytesPerSec = readLittleEndianInt(input);
+			writeLittleEndianInt(output, nAvgBytesPerSec);
+			ctx.nBlockAlign = Short.reverseBytes(input.readShort());
+			ctx.wBitsPerSample = Short.reverseBytes(input.readShort());
+			output.writeInt((Short.reverseBytes((short) ctx.nBlockAlign) << 16) | Short.reverseBytes((short) ctx.wBitsPerSample));
+			ctx.hasfmt = true;
+			if(size > 16) {
+				short cbSize = Short.reverseBytes(input.readShort());
+				if(cbSize + 18 != size) {
+					throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "fmt chunk size is invalid");
+				}
+				output.writeShort(Short.reverseBytes(cbSize));
+			}
+			if(size > 18) {
+				// wValidBitsPerSample, dwChannelMask, and SubFormat GUID
+				passthroughBytes(input, output, 22);
+			}
+			// Further checks
+			if((ctx.format == WAVE_FORMAT_ALAW || ctx.format == WAVE_FORMAT_MULAW) && ctx.wBitsPerSample != 8) {
+				throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "Unexpected bits per sample value");
+			}
+			return;
+		} else if(!ctx.hasfmt) {
+			throw new DataFilterException(l10n("invalidTitle"), l10n("invalidTitle"), "Unexpected header chunk was encountered, instead of fmt chunk");
+		}
+		if(ID[0] == 'd' && ID[1] == 'a' && ID[2] == 't' && ID[3] == 'a') {
+			if(ctx.format == WAVE_FORMAT_PCM || ctx.format == WAVE_FORMAT_IEEE_FLOAT || ctx.format == WAVE_FORMAT_ALAW || ctx.format == WAVE_FORMAT_MULAW) {

```

Comment:
This condition will always evaluate to true (ctx.format is only written in line 51 and checked immediately after)